Marlin/u8g_fontutf8.c

```diff
@@ -0,0 +1,403 @@
+/**
+ * @file    fontutf8.c
+ * @brief   font api for u8g lib
+ * @author  Yunhui Fu (yhfudev@gmail.com)
+ * @version 1.0
+ * @date    2015-02-19
+ * @copyright GPL/BSD
+ */
+
+#include <string.h>
+
+#include "fontutils.h"
+#include "u8g.h"
+
+#include "u8g_fontutf8.h"
+
+#define FALSE 0
+#define TRUE  1
+
+//#if DEBUG
+//#include <stdio.h>
+//#include <stdlib.h>
+//#define assert(a) if (!(a)) {printf("Assert: " # a); exit(1);}
+//#define TRACE(fmt, ...) fprintf (stdout, "[%s()] " fmt " {ln:%d, fn:" __FILE__ "}\n", __func__, ##__VA_ARGS__, __LINE__)
+//#else
+//#define assert(a)
+//#define TRACE(...)
+//#endif
+
+////////////////////////////////////////////////////////////
+#define font_t void
+
+/**
+ * @brief the callback function to draw somthing
+ *
+ * @param userdata : User's data
+ * @param msg : the u8g's string
+ * @param fnt_current : the font
+ *
+ * @return 0 on success, 1 to force quit, <0 on error
+ *
+ * Get the screen pixel length of a ROM UTF-8 string
+ */
+typedef int (* fontgroup_cb_draw_t)(void *userdata, font_t *fnt_current, const char *msg);
+
+//extern int fontgroup_init (font_group_t * root, uxg_fontinfo_t * fntinfo, int number);
+//extern int fontgroup_drawstring (font_group_t * group, font_t *fnt_default, const char *utf8_msg, void * userdata, fontgroup_cb_draw_t cb_draw);
+//extern uxg_fontinfo_t * fontgroup_first (font_group_t * root);
+
+
+////////////////////////////////////////////////////////////
+/* return v1 - v2 */
+static int
+fontinfo_compare (uxg_fontinfo_t * v1, uxg_fontinfo_t * v2)
+{
+    assert (NULL != v1);
+    assert (NULL != v2);
+    if (v1->page < v2->page) {
+        return -1;
+    } else if (v1->page > v2->page) {
+        return 1;
+    }
+    if (v1->end < v2->begin) {
+        return -1;
+    } else if (v1->begin > v2->end) {
+        return 1;
+    }
+    return 0;
+}
+
+/*"data_list[idx] - *data_pin"*/
+static int
+pf_bsearch_cb_comp_fntifo_pgm (void *userdata, size_t idx, void * data_pin)
+{
+    uxg_fontinfo_t * fntinfo = (uxg_fontinfo_t *) userdata;
+    uxg_fontinfo_t localval;
+    memcpy_P (&localval, fntinfo + idx, sizeof (localval));
+    return fontinfo_compare (&localval, data_pin);
+}
+
+typedef struct _font_group_t {
+    uxg_fontinfo_t * m_fntifo;
+    int m_fntinfo_num;
+} font_group_t;
+
+static int
+fontgroup_init (font_group_t * root, uxg_fontinfo_t * fntinfo, int number)
+{
```

Comment:
Do you have a "indent" settings(BASH command indent) or other tools to re-format the source code? It would be hard to change the format by manual, and may make mistakes during the formatting.